lib/spack/docs/tutorial_buildsystems.rst

```diff
@@ -0,0 +1,351 @@
+usng the build() and install() methods.
+
+.. literalinclude:: tutorial/examples/Makefile/3.package.py
+   :language: python
+
+In our build stage we are able to provide an argument to make whether we
+decide to build with tbb support or not. For the install, phase we provide
+the prefix and then invoke make install. 
+
+Through this example, we are able to see that we can change the Makefile through
+our edit method, as well as invoke build, and install to specify our build
+and install targets, respectively.
+
+-------------------
+Autotools
+-------------------
+
+As mentioned earlier, Autotools consist of three parts:  autoconf, automake,
+and libtool. Your typical build incantation will consist of the following:
+
+.. code-block:: console
+
+    ./configure --prefix=/path/to/install_dir
+    make
+    make install
+
+Although it is possible to write these packages using the vanilla Package 
+class, Spack has an Autotools class that automatically sets up common 
+arguments to configure. In the Autotools class we can override four phases: 
+autoreconf, configure, build, and install. Autoreconf is only necessary if 
+there is no configure script present and one has to be generated using. 
+Most packagers will only need to override the helper method 
+configure_args since Spack provides sensible defaults for each phase. 
+
+Let's look at the mpileaks package.py file that we worked on earlier:
+
+
+.. code-block:: console
+    
+    $ spack edit mpileaks
+
+Notice that mpileaks is a Package class but uses the Autotools build system 
+(the configure, make, make install incantation gives us a hint). Although
+this package is acceptable lets make this into an AutotoolsPackage class and
+simplify it further.
+
+.. literalinclude:: tutorial/examples/Autotools/0.package.py
+
+We first inherit from the AutotoolsPackage class. 
+
+We can keep the install method but we can simplify this further by adding configure_args:
+
+.. literalinclude:: tutorial/examples/Autotools/1.package.py
+
+A couple of things to note here: We got rid of "prefix=" + prefix, and also
+the calls to make() and make("install"). So not only does our package.py
+file look simpler, but we can also get rid of repeated code and not have to
+worry about having to invoke configure and make, make install. Even better
+is that we have allowed further control over what targets we choose to 
+build during our make phase as well as additional functionality. For example,
+if we wish to add a check or test phase in our install we can add:
+
+.. code-block:: python
+    
+    def check()
+
+to our package.py file. 
+
+---------------
+CMake
+---------------
+
+CMake_ is another common build system that has been gaining popularity. It works
+in a similar manner to autotools but with differences in variable names,
+the number of configuration options available, and the handling of shared libraries.
+Most CMake projects are built out of source which means that there is a dedicated
```

Comment:
Does this sentence improve the tutorial? It's true, but I think it might distract people by making them think it's relevant to packaging.